pandas/core/indexes/base.py

```diff
@@ -2929,7 +2940,8 @@ def difference(self, other):
         self._assert_can_do_setop(other)
 
         if self.equals(other):
-            return self._shallow_copy([])
+            # pass an empty array with the appropriate dtype
+            return self._shallow_copy(self[:0])
```

Comment:
did you add a test for this?

Reply:
Or this was needed due to the change in shallow_copy?

Reply:
This is just because `_shallow_copy` calls `_simple_new`, and being "simple" should not be getting a list.

Reply:
No, I did not add a test, but can confirm this line is hit in the tests.